source/common/http/http3/conn_pool.cc

```diff
@@ -28,7 +28,19 @@ uint32_t getMaxStreams(const Upstream::ClusterInfo& cluster) {
 ActiveClient::ActiveClient(Envoy::Http::HttpConnPoolImplBase& parent,
                            Upstream::Host::CreateConnectionData& data)
     : MultiplexedActiveClientBase(parent, getMaxStreams(parent.host()->cluster()),
-                                  parent.host()->cluster().stats().upstream_cx_http3_total_, data) {
+                                  parent.host()->cluster().stats().upstream_cx_http3_total_, data),
+      async_connect_callback_(parent_.dispatcher().createSchedulableCallback([this]() {
+        if (state() == Envoy::ConnectionPool::ActiveClient::State::CONNECTING) {
+          codec_client_->connect();
+        }
+      })) {
+  ASSERT(codec_client_);
+  if (dynamic_cast<CodecClientProd*>(codec_client_.get()) == nullptr) {
+    ASSERT(Runtime::runtimeFeatureEnabled(
+        "envoy.reloadable_features.postpone_h3_client_connect_till_enlisted"));
+    // Hasn't called connect() yet, schedule one for the next event loop.
```

Comment:
let's comment why we defer, so folks don't have to dig up git history for it

Reply:
done